### ISSUE TITLE
creating a Queue from a multiprocessing.Manager sometimes crashes

### ISSUE THREAD

**Issue by pyinstaller-tickets-migration** [pyinstaller/pyinstaller#486]
**Original date**: 2012/01/04
**Original reporter**: *aargri AND gmail DOT  COOM * @agrif

The following test program will crash for me about once every six runs. It seems like pyinstaller is having trouble extracting a module from an archive, but since this doesn't happen on *every* run, this bug is a little scary.

```#!python
#!/usr/bin/python
import multiprocessing

if __name__ == "__main__":
    multiprocessing.freeze_support()
    manager = multiprocessing.Manager()
    queue = manager.Queue()
```

This is the full traceback. Note that the actual error string ("incorrect header check" in this case) differs from run to run; I have also seen "invalid literal/lengths set" and "invalid bit length repeat".

```
Traceback (most recent call last):
  File "<string>", line 8, in <module>
  File "/Users/overviewer/agrif/tests/build/pyi.darwin/manager_test/outPYZ1.pyz/multiprocessing.managers", line 634, in temp
  File "/Users/overviewer/agrif/tests/build/pyi.darwin/manager_test/outPYZ1.pyz/multiprocessing.managers", line 532, in _create
  File "/Users/overviewer/agrif/tests/build/pyi.darwin/manager_test/outPYZ1.pyz/multiprocessing.connection", line 140, in Client
  File "/Users/overviewer/agrif/tests/build/pyi.darwin/manager_test/outPYZ1.pyz/multiprocessing.connection", line 369, in answer_challenge
  File "/Users/overviewer/agrif/pyinstaller/PyInstaller/loader/iu.py", line 449, in importHook
    mod = _self_doimport(nm, ctx, fqname)
  File "/Users/overviewer/agrif/pyinstaller/PyInstaller/loader/iu.py", line 518, in doimport
    mod = director.getmod(nm)
  File "/Users/overviewer/agrif/pyinstaller/PyInstaller/loader/iu.py", line 309, in getmod
    mod = owner.getmod(nm)
  File "/Users/overviewer/agrif/pyinstaller/PyInstaller/loader/archive.py", line 426, in getmod
    rslt = self.pyz.extract(nm)
  File "/Users/overviewer/agrif/pyinstaller/PyInstaller/loader/archive.py", line 336, in extract
    obj = zlib.decompress(obj)
zlib.error: Error -3 while decompressing data: incorrect header check
```

Line 369 in `multiprocessing.connection`, the line that triggers this apparent import error, is `import hmac` on my system. I should also add that in the larger program that first triggered this bug, sometimes it would raise a different exception about an invalid message digest, as though the hmac module did not import correctly.

I am running pyinstaller SVN, revision 1899, on OS X 10.6.2 with Python 2.6.1 and XCode 3.2. If you need any more information I would be happy to provide it.

**Comment by htgoebel** [comment]
@agrif: Hi, this is an old PyInstaller issue by you. Can you please verify it is still an issue, otherwise close it. Thanks.

If it is still an issue, would you please be so kind and check if #1161 solves it. Thanks

**Comment by agrif** [comment]
I've run the old test program about 2300 times on my machine with no crashes, so I'm about 99.7% confident it'll fail less than 0.3% of the time now. I think the bug's gone.

I lost the exact machine this bug happened on a while ago, so I tested this on pyinstaller 2.1 (from pip) on OSX 10.9.5 with python 2.7.9 and xcode 6.0.

I'd close the issue, but github doesn't think I created it so it won't let me.

**Comment by htgoebel** [comment]
Thanks for testing. Did you test "plain" pyinstaller or with #1161 applied?

**Comment by agrif** [comment]
Plain pyinstaller only, no patches.

**Comment by htgoebel** [comment]
Merged #1161, so I can close this issue new.
